clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java

```
@@ -132,18 +135,30 @@ CompletionStage<Void> bootstrapFromBlobs(
     AbstractStorageEngine storageEngine = storageService.getStorageEngine(kafkaTopic);
     if (storageEngine != null && storageEngine.containsPartition(partitionId)) {
       storageEngine.dropPartition(partitionId, false);
+      LOGGER.info(
+          "Due to storage engine contains this partition, clean up the offset and delete partition folder for topic {} partition {} before bootstrap from blob transfer",
+          kafkaTopic,
+          partitionId);
+    }
+
+    // double check that the partition folder is not existed before bootstrapping from blobs transfer,
+    // as the partition should not exist due to previous dropPartition call.
+    String partitionFolder = RocksDBUtils.composePartitionDbDir(serverConfig.getRocksDBPath(), kafkaTopic, partitionId);
+    File partitionFolderDir = new File(partitionFolder);
+    if (partitionFolderDir.exists()) {
+      LOGGER.error(
```

Shall we fail fast in this case?

---

I intentionally printed it out and did not fail fast in order to reproduce the issue and verify whether it is caused by files not being cleaned up. Once this assumption is confirmed, we can submit a new PR to fail the blob transfer and allow it to fall back to Kafka ingestion.